### PR TITLE
Adds syslog+tls:// drain type

### PR DESCRIPTION
**Pull request by archaelus**
This is some work in progress code for a TLS syslog drain implementation. It's largely a clone and modify of the tcpsyslog implementation, though maybe it'd be a better plan to go with a two process design like the http drain stuff.

Currently missing:
* any testing at all
* ssl cert validation

### CODE REVIEW

**Comment by archaelus** [comment]
Things of note: 
* old rebar (v2.x) is insanely slow - this should be upgraded to rebar3
* There are lots of incompatibilities with erlang R17.5 due to 'now()' use.

**Comment by archaelus** [comment]
This implementation should add logplex_tls that defines the SSL/TLS connection options (certificate validation etc) for using the OTP ssl library. Then the new syslog.tls code should use it, as should the logplex http drain code (would need passing down into hackney land iirc).

File: src/logplex_tlssyslog_drain.erl [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+                %%     Node = get_node(Process),
+                %%     exit({nodedown, Node});
+                %% {'DOWN', Ref, _, _, Reason} ->
+                %%     exit(Reason)
+                {sslsocket, _, Pid} = Sock,
+                MRef = erlang:monitor(process, Pid),
+                erlang:send(Pid,
+                            {'$gen_sync_all_state_event',
+                             {self(), MRef},
+                             {application_data, iolist_to_binary(Data)}}),
```
Comment (voidlock) [inline src/logplex_tlssyslog_drain.erl]: Is there a reason why this monitoring and intimate knowledge of the bowels of the ssl connection logic is required here?

Comment (archaelus) [inline src/logplex_tlssyslog_drain.erl]: Unfortunately yeah - the process design for tcp and now TLS drains requires that no blocking operations happen in the process handling the log channel buffer (which is combined with the drain itself for these two drain types - unlike the http drain). The SSL library doesn't expose an async send + get-message-on-result API, so we dig into the internals to get an async message interface.

The alternative is to split the sender and message buffering. That may be an easier to maintain approach.

**Pull request merged**